openpgp/packet/private_key.go

```diff
@@ -404,7 +404,18 @@ func (pk *PrivateKey) Decrypt(passphrase []byte) error {
 		data = data[:len(data)-2]
 	}
 
-	return pk.parsePrivateKey(data)
+	err := pk.parsePrivateKey(data)
+	if err != nil {
+		return err
+	}
+
+	if !pk.Dummy() {
+		pk.s2kType = S2KNON
+		pk.s2k = nil
+		pk.Encrypted = false
```

Comment:
All of the functions that `parsePrivateKey` calls already set `pk.Encrypted = false`, so this is a bit redundant; that being said, I agree that this is the more logical place to do it, so you could also remove all of those. But as you prefer :)